pipeline/frontend/yaml/compiler/compiler.go

```diff
@@ -187,7 +187,7 @@ func (c *Compiler) Compile(conf *yaml_types.Workflow) (*backend_types.Config, er
 			step := c.createProcess(name, container, backend_types.StepTypeClone)
 
 			// only inject netrc if it's a trusted repo or a trusted plugin
-			if !c.netrcOnlyTrusted || c.trustedPipeline || (container.IsPlugin() && container.IsTrustedCloneImage()) {
+			if !c.netrcOnlyTrusted || c.trustedPipeline || (c.netrcOnlyTrusted && container.IsPlugin() && container.IsTrustedCloneImage()) {
```

Comment:
This change is not needed as it should be true:
- if netrc passing is allowed for not trusted images
- if repo is trusted
- if its a plugin and using a trusted clone image (regardless of the netrc option)